src/widgets/immunizations/immunizations-form.component.tsx

```diff
@@ -50,19 +50,13 @@ export function ImmunizationsForm(props: ImmunizationsFormProps) {
   const currentUser = useSessionUser();
 
   useEffect(() => {
-    if (formState.vaccinationDate) {
-      updateSingle("enableCreateButtons", true);
-    } else {
-      updateSingle("enableCreateButtons", false);
-    }
+    const enableCreate = !!formState.vaccinationDate;
+    updateSingle("enableCreateButtons", enableCreate);
```

Comment:
Why is this still in an effect? It should not be a side-effect. Just compute it once.

Instead of

```
useEffect(() => {
  // ...
}, [...]);
```

use

```ts
const enableCreate = !!formState.vaccinationDate && formState.viewEditForm && formState.formChanged;
```

Reply:
I implemented this feedback in the latest commit after I got the understanding right. It has been changed in the latest commit [here](https://github.com/openmrs/openmrs-esm-patient-chart-widgets/pull/118/commits/de577dce3fae84327090b88db7ab8c234ce18945).